pkg/metrics/metrics.go

```diff
@@ -89,14 +94,14 @@ func (mm *manager) RegisterMetricsCollector(targetPath, podNamespace, podName, b
 	}
 
 	podUID, volumeName, _ := util.ParsePodIDVolumeFromTargetpath(targetPath)
-	promFilePath := emptyDirBasePath + metricsFileName
-	c := NewTextFileCollector(promFilePath, podUID, volumeName, map[string]string{
+	c := NewTextFileCollector(emptyDirBasePath, podUID, volumeName, map[string]string{
 		"pod_name":       podName,
 		"namespace_name": podNamespace,
 		"volume_name":    volumeName,
 		"bucket_name":    bucketName,
+		"pod_uid":        podUID,
 	})
-	if err := mm.registry.Register(c); err != nil && !strings.Contains(err.Error(), prometheus.AlreadyRegisteredError{}.Error()) {
+	if err := mm.registry.Register(c); err != nil && !errors.Is(err, prometheus.AlreadyRegisteredError{}) {
```

Comment:
We don't have test coverage for this line. Please manually check log to make sure we don't see this error message in re-publish calls.

Reply:
From the CSI driver log, I see the error message `duplicate metrics collector registration attempted`. It means the change cannot skip the `AlreadyRegisteredError`.

Reply:
seeing the same thing. we can revert this change.

Reply:
@kislaykishore I afraid we need to reverse this change and use `strings.Contains`. Seems like the error is wrapped so we cannot directly use `errors.Is` to check if the error is the same.